### PR TITLE
Add newer api endpoints to docs

### PR DESCRIPTION
**Pull request by catalinstoica**
Adding newer api endpoints docs:
- Ranking genres metadata
- Ranking types metadata
- App availability
- App publishers

### CODE REVIEW

File: source/includes/ranking-genres-metadata.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+{"name":"Music & Audio","genre_id":"MUSIC_AND_AUDIO"}
+{"name":"Auto & Vehicles","genre_id":"AUTO_AND_VEHICLES"}
+{"parent_id":"FAMILY","name":"Family Brain Games","genre_id":"FAMILY_BRAINGAMES"}
+
+{"name":"Books","genre_id":"6018"}
+{"name":"Stickers","genre_id":"6025"}
+{"parent_id":"6021","name":"Health, Mind & Body","genre_id":"13017"}
+```
+
+Request all ranking genres by date.
```
Comment (lknix) [inline source/includes/ranking-genres-metadata.md]: Maybe we should add: (categories)  after genres, because Google Play uses it.

Comment (lknix) [reply source/includes/ranking-genres-metadata.md]: test

Comment (lknix) [reply source/includes/ranking-genres-metadata.md]: test

File: source/includes/app-availability.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+```
+{"app_id":"com.wikia.singlewikia.dragonvale","countries":["FR","DE","JP","US"]}
+{"app_id":"com.BLI.CrossTheRoad","countries":["FR","JP","US"]}
+{"app_id":"com.singlecase.app","countries":["FR","DE","JP","US"]}
+
+{"app_id":1358954496,"countries":["MO","HK","CN","TW"]}
+{"app_id":440401960,"countries":["US"]}
+{"app_id":1350566056,"countries":["US","FR"]}
+```
+Get a list of all the countries where an app is available.
```
Comment (lknix) [inline source/includes/app-availability.md]: where the app is available.

File: source/includes/app-publishers.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+{"name":"andavis","url":"http:\/\/www.andavis.de","id":"andavis","address":"","email":"apps@andavis.de"}
+{"name":"Air Cab Ltd","url":"http:\/\/aircab.vn","id":"Air Cab Ltd","address":"","email":"aircab.vn@gmail.com"}
+{"name":"Right Pulse Works","url":"","id":"Right Pulse Works","address":"","email":"rightpulseapps@gmail.com"}
+
+{"url":"http:\/\/www.ozogluozbay.com.tr\/","name":"mustafa diler","id":"1118647768"}
+{"url":"http:\/\/blink.sh","name":"Carlos Cabanero","id":"1156707511"}
+{"url":"https:\/\/papplication.github.io\/site\/","name":"Oliver Papp","id":"896476189"}
+
+```
+Get a list of all publisher IDs and details that AppMonsta knows about.
```
Comment (lknix) [inline source/includes/app-publishers.md]: Get a list of all publishers and their information that AppMonsta knows about.

File: source/includes/ranking-types-metadata.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+{"rank_id":"topfreeapplications","name":"Top Free"}
+{"rank_id":"newapplications","name":"New Applications"}
+{"rank_id":"newgameswelove","name":"New Games We Love RSS"}
+
+{"rank_id":"apps_movers_shakers","name":"Trending Apps"}
+{"rank_id":"apps_tablet_featured","name":"Staff Picks for Tablet"}
+{"rank_id":"apps_topselling_new_free","name":"Top New Free"}
+```
+
+Request all ranking genres by date.
```
Comment (lknix) [inline source/includes/ranking-types-metadata.md]: This is wrong.

File: source/includes/rankings.md [hunk 5]
```diff
@@ -216,3 +217,568 @@ Header            | Description
...
+ "genre_id":"LIFESTYLE"}
+
+{"ranks":["479516143","763692274","1396301917","640364616",...],
+ "country":"US",
+ "rank_id":"toppaidapplications",
+ "genre_id":"7002"}
+```
+
+
+Request aggregate rankings by date and by country.
```
Comment (lknix) [inline source/includes/rankings.md]: Add: This call returns records with ranked app ids per country and per genre / type (not other meta-data). It's ideal if you're only interested in app rank positions.

NOTE: App ids in the list are sorted according to their rank (from the top to the bottom). First app in the list is the top ranking app.

File: source/includes/rankings.md [hunk 6]
```diff
@@ -216,3 +217,568 @@ Header            | Description
...
+{"name":"Music & Audio","genre_id":"MUSIC_AND_AUDIO"}
+{"name":"Auto & Vehicles","genre_id":"AUTO_AND_VEHICLES"}
+{"parent_id":"FAMILY","name":"Family Brain Games","genre_id":"FAMILY_BRAINGAMES"}
+
+{"name":"Books","genre_id":"6018"}
+{"name":"Stickers","genre_id":"6025"}
+{"parent_id":"6021","name":"Health, Mind & Body","genre_id":"13017"}
+```
+
+Request all ranking genres by date.
```
Comment (lknix) [inline source/includes/rankings.md]: Request all available ranking genres (also know as categories) by date. If you use (link to aggregate API call) then you this call will map genre IDs to human readable genre names.

**Comment by lknix** [comment]
Fix all typos e.g. "id" -> "ID" and similar.

File: source/includes/rankings.md [hunk 7]
```diff
@@ -216,3 +217,568 @@ Header            | Description
...
+{"rank_id":"topfreeapplications","name":"Top Free"}
+{"rank_id":"newapplications","name":"New Applications"}
+{"rank_id":"newgameswelove","name":"New Games We Love RSS"}
+
+{"rank_id":"apps_movers_shakers","name":"Trending Apps"}
+{"rank_id":"apps_tablet_featured","name":"Staff Picks for Tablet"}
+{"rank_id":"apps_topselling_new_free","name":"Top New Free"}
+```
+
+Request all ranking types by date.
```
Comment (lknix) [inline source/includes/rankings.md]: OK, we need to really explain what this means.

File: source/includes/rankings.md [hunk 8]
```diff
@@ -216,3 +217,568 @@ Header            | Description
 <aside class="notice">
 This is a bulk API call. Bulk API calls return one record per line.
 </aside>
+
+## Aggregate rankings
```
Comment (lknix) [inline source/includes/rankings.md]: Aggregated rankings

File: source/includes/rankings.md (hunk 5 above)
Comment (lknix) [reply source/includes/rankings.md]: typo: aggregated rankings

Comment (lknix) [reply source/includes/rankings.md]: and yes API call has `aggregate` in the path (which is right - it's used as a noun whereas aggregated is used as a verb).

**Comment by lknix** [comment]
These explanations are not very helpful:

```
genre_id | Genre id (string). We force “overall” id value for top level ranking lists.
-- | --
rank_id | Ranking list type which serves also as identifier. Set by stores.

```

**Comment by lknix** [comment]
More typos like: ` (array of strings for consistency between iTunes and google).`

**Comment by lknix** [comment]
It should be: iTunes and Google Play stores.

File: source/includes/rankings.md (hunk 7 above)
Comment (lknix) [reply source/includes/rankings.md]: Ranking types are ranking list type (names) as defined by the Google Play or iTunes store. If you use (aggregated ranks API call link), then you can use this API call to map ranking types to their human-readable names.

**Comment by lknix** [comment]
For general Rankings API call add: Request all rankings by date and by country. This call returns ranking records with all the available meta-data from the ranking list page.

**Comment by lknix** [comment]
@catalinstoica there's plenty to fix here before we can merge.

**Comment by catalinstoica** [comment]
@lknix added bunch of commits and this seems fine to me; can I get a second review when you get the chance? Thanks!

File: source/includes/rankings.md [hunk 9]
```diff
@@ -381,7 +381,11 @@ curl_close($ch);
 ```
 
 
-Request aggregate rankings by date and by country.
+Request aggregated rankings by date and by country. This call returns records with ranked app ID's per country and per genre/type (not other meta-data). It's ideal if you're only interested in app rank positions.
```
Comment (lknix) [inline source/includes/rankings.md]: Typos, fix: This call returns records with ranked app IDs per country and per genre/category (without other meta-data).

File: source/includes/rankings.md [hunk 10]
```diff
@@ -381,7 +381,11 @@ curl_close($ch);
...
+
+<aside class="notice">
+App ID's in the list are sorted according to their rank (from the top to the bottom). First app in the list is the top ranking app.
```
Comment (lknix) [inline source/includes/rankings.md]: Typo: App IDs in the list are sorted according to their rank ...

**Comment by lknix** [comment]
Rename: `Genres metadata` -> `Ranking genres`, `List types metadata` -> `Ranking list types`

**Comment by lknix** [comment]
Bug: genres API endpoint code sample has a country code parameter (check others too)

File: source/includes/rankings.md [hunk 11]
```diff
@@ -572,7 +572,7 @@ curl_close($ch);
 {"parent_id":"6021","name":"Health, Mind & Body","genre_id":"13017"}
 ```
 
-Request all available ranking genres (also known as categories) by date. If you use [aggregated](#aggregated-rankings) API call then this call will map genre IDs to human readable genre names.
+Request all available ranking genres (also known as categories) by date. If you use [aggregated](#aggregated-rankings) API call, then this call will map genre IDs to human readable genre names.
```
Comment (lknix) [inline source/includes/rankings.md]: human-readable

File: source/includes/rankings.md [hunk 12]
```diff
@@ -757,7 +757,7 @@ curl_close($ch);
 {"rank_id":"apps_topselling_new_free","name":"Top New Free"}
 ```
 
-Request all ranking types by date.
+Request all ranking types by date. Ranking types are ranking list types (names) as defined by the Google Play or iTunes store. If you use [aggregated](#aggregated) ranks API call, then this call will map ranking types to their human-readable names.
```
Comment (lknix) [inline source/includes/rankings.md]: You probably mean: "ranking list types" (don't shorten it).

File: source/includes/rankings.md [hunk 13]
```diff
@@ -177,7 +177,7 @@ curl_close($ch);
 }
 ```
 
-Request all rankings by date and by country.
+Request all rankings by date and by country. This call returns ranking records with all the available meta-data from the ranking list page.
```
Comment (lknix) [inline source/includes/rankings.md]: the store's ranking list page.

File: source/includes/rankings.md [hunk 14]
```diff
@@ -405,8 +405,8 @@ Parameter         | Required | Value
 Field               | Description
 ------------------- | -----------
 **country**         | The country this ranking list is for; 2 letter country code.
-**genre_id**        | Genre ID (string). We force “overall” ID value for top level ranking lists.
-**rank_id**         | Ranking list type which serves also as identifier. Set by stores.
+**genre_id**        | String representing rankin list genre/category, as returned by the store. We force “overall” ID value for top level ranking lists.
```
Comment (lknix) [inline source/includes/rankings.md]: typo: String representing ranking genre/category, as defined by the store.
typo2: I think you're not using regular double quotes: "" for overall.

**Comment by lknix** [comment]
This is starting to look really good!

**Comment by catalinstoica** [comment]
> 
> 
> Bug: genres API endpoint code sample has a country code parameter (check others too)

Only general and aggregated seem to have country code param; point me to that sample if still there pls...

File: source/includes/rankings.md (hunk 14 above)
Comment (catalinstoica) [reply source/includes/rankings.md]: Fixed all typos in https://github.com/AppMonsta/slate/commit/615c1afe0924c8b14ebb98b2843819b00445175f

**Comment by catalinstoica** [comment]
Removed itunes response samples in https://github.com/AppMonsta/slate/commit/545eaf47098a3c6605111a92b5c0883b070aacbb

**Comment by lknix** [comment]
LGTM! go ahead merge and then add it to scrappstore repo + deploy to staging. If it looks good on staging, go ahead and redeploy production!

**Pull request merged**